src/tab-list.js

```diff
@@ -0,0 +1,76 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class TabList extends React.Component {
+  constructor() {
+    super();
+    this.state = { focus: false };
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.handlers = {
+      ArrowLeft: this.previous.bind(this),
+      ArrowRight: this.next.bind(this),
+    };
+  }
+
+  wrapIndex(index) {
+    const count = React.Children.count(this.props.children);
+    return (index + count) % count;
+  }
+
+  handleBlur() {
+    this.setState({ focus: false });
+  }
+
+  handleKeyPress(event) {
+    const handler = this.handlers[event.key] || (() => {});
+    handler();
+  }
+
+  previous() {
+    const newIndex = this.wrapIndex(this.props.activeIndex - 1);
+    this.setState({ focus: true });
+    this.props.onActivateTab(newIndex)
+  }
+
+  next() {
+    const newIndex = this.wrapIndex(this.props.activeIndex + 1);
+    this.setState({ focus: true });
+    this.props.onActivateTab(newIndex)
+  }
+
+  render() {
+    const { accessibleId, activeIndex, children, className, onActivateTab } = this.props;
+
+    return (
+      <ul className={className} onBlur={this.handleBlur} onKeyUp={this.handleKeyPress} role="tablist">
+        {React.Children.map(children, (child, index) => {
+          const active = index === activeIndex;
```

Comment:
Food for thought: should this component handle disabled tabs as well? I would imagine that a disable tab would just get skipped.